dataloader/dataloader.hpp

```diff
@@ -0,0 +1,245 @@
+   *                              column.
+   * @param augmentation Vector strings of augmentations supported by mlpack.
+   * @param augmentationProbability Probability of applying augmentation to a particular cell.
+   */
+  void LoadCSV(const std::string &datasetPath,
+               const bool loadTrainData = true,
+               const bool shuffle = true,
+               const double ratio = 0.75,
+               const bool useScaler = false,
+               const bool dropHeader = false,
+               const int startInputFeatures = -1,
+               const int endInputFeatures = -1,
+               const int startPredictionFeatures = -1,
+               const int endPredictionFeatures = -1,
+               const std::vector<std::string> augmentation =
+                   std::vector<std::string>(),
+               const double augmentationProbability = 0.2);
+
+  //! Get the Training Dataset.
+  DatasetX TrainX() const { return trainX; }
+
+  //! Modify the Training Dataset.
+  DatasetX& TrainX() { return trainX; }
+
+  //! Get the Training Dataset.
+  DatasetY TrainY() const { return trainY; }
+  //! Modify the Training Dataset.
+  DatasetY& TrainY() { return trainY; }
+
+  //! Get the Test Dataset.
+  DatasetX TestX() const { return testX; }
+  //! Modify the Test Dataset.
+  DatasetX& TestX() { return testX; }
+
+  //! Get the Test Dataset.
+  DatasetY TestY() const { return testY; }
+  //! Modify the Training Dataset.
+  DatasetY& TestY() { return testY; }
+
+  //! Get the Validation Dataset.
+  DatasetX ValidX() const { return validX; }
+  //! Modify the Validation Dataset.
+  DatasetX& ValidX() { return validX; }
+
+  //! Get the Validation Dataset.
+  DatasetY ValidY() const { return validY; }
+  //! Modify the Validation Dataset.
+  DatasetY& ValidY() { return validY; }
+
+  //! Get the Scaler.
+  ScalerType Scaler() const { return scaler; }
+  //! Modify the Sclaer.
+  ScalerType& Scaler() { return scaler; }
+
+ private:
+  /**
+   * Downloads and checks hash for given dataset.
+   */
+  void DownloadDataset(const std::string& dataset)
+  {
+    if (!Utils::PathExists(datasetMap[dataset].trainPath))
+    {
+      Utils::DownloadFile(datasetMap[dataset].trainDownloadUrl,
+          datasetMap[dataset].trainPath, dataset + "_training_data.",
+          false);
+
+      if (!Utils::CompareCRC32(datasetMap[dataset].trainPath,
+          datasetMap[dataset].trainHash))
+      {
+        mlpack::Log::Fatal << "Corrupted Training Data for " <<
+            dataset << " Downloaded." << std::endl;
+      }
+    }
+    if (!Utils::PathExists(datasetMap[dataset].testPath))
+    {
+      Utils::DownloadFile(datasetMap[dataset].trainDownloadUrl,
+          datasetMap[dataset].testPath, dataset + "_testing_data.",
+          false);
+
+      if (!Utils::CompareCRC32(datasetMap[dataset].testPath,
+          datasetMap[dataset].testHash))
+        mlpack::Log::Fatal << "Corrupted Testing Data for " <<
```

Comment:
I would use `{}` here to make this easier to read.

Reply:
Agreed.